src/js/libcs/OpImageDrawing.js

```diff
@@ -347,12 +347,102 @@ evaluator.allimages$0 = function() {
     return List.turnIntoCSList(lst);
 };
 
-evaluator.cameravideo$0 = function() {
+evaluator.cameravideo$0 = function(args, modifs) {
+    var maximal = true; //use maximal as default (if no other modifier is given)
+    var constraints = {};
+
+    if (modifs.resolution !== undefined) {
+        maximal = false;
+        var val = evaluate(modifs.resolution);
+        if (val.ctype === 'string' && val.value === 'maximal') maximal = true;
+        else {
+            var width = null;
+            var heightorratio = null;
+
+            if (val.ctype === 'number') {
+                width = val.value.real;
+            } else if (List.isNumberVector(val).value && (val.value.length === 2)) {
+                width = val.value[0].value.real;
+                heightorratio = val.value[1].value.real;
+            } else return nada;
```

Comment:
Usually we ignore modifiers of invalid type, instead of failing the whole function call. Is there a reason to do things differently here?